src/main.ts

```diff
@@ -116,80 +137,110 @@ async function run(): Promise<void> {
-      console.log(`###### TOUCHED LOCKS FOUND ######`)  //DEBUG
-      console.log(pr_diff_body.data.match(search_locked_lines_regexp))  //DEBUG
+    const search_locked_lines_regexp: RegExp = /🔒.*(\n^[+|-].*)|^[+|-].*🔒/gm
+    if (pr_diff_body.data.match(search_locked_lines_regexp) !== null) {
+      console.log(`###### TOUCHED LOCKS FOUND ######`) //DEBUG
+      console.log(pr_diff_body.data.match(search_locked_lines_regexp)) //DEBUG
       CUSTOM_REVIEW_REQUIRED = true
-      var approvers: string[] = []
-      await combineUsersTeams(octokit, context, organization, pr_owner, [], ['pr-custom-review-team']).then(value => {
-        approvers = value
+      var approvers: string[] = await combineUsersTeams(
+        octokit,
+        context,
+        organization,
+        pr_owner,
+        [],
+        ["pr-custom-review-team"],
+      )
+      final_approval_groups.push({
+        name: "LOCKS TOUCHED",
+        min_approvals: 2,
+        users: [],
+        teams: ["pr-custom-review-team"],
+        approvers: approvers,
       })
-      final_approval_groups.push({ name: 'LOCKS TOUCHED', min_approvals: 2, users: [], teams: ['pr-custom-review-team'], approvers: approvers })
-      console.log(final_approval_groups)  //DEBUG
+      console.log(final_approval_groups) //DEBUG
       pr_status_messages.push(`LOCKS TOUCHED review required`)
     }
 
     // Read values from config file if it exists
     console.log(`###### CONFIG FILE EVALUATION ######`) //DEBUG
     var config_file_contents: any = ""
-    if (fs.existsSync(core.getInput('config-file'))) {
-      const config_file = fs.readFileSync(core.getInput('config-file'), 'utf8')
+    if (fs.existsSync(core.getInput("config-file"))) {
+      const config_file = fs.readFileSync(core.getInput("config-file"), "utf8")
       config_file_contents = YAML.parse(config_file)
 
       for (const approval_group of config_file_contents.approval_groups) {
-        console.log(`approval_group: ${approval_group.name}`)  //DEBUG
+        console.log(`approval_group: ${approval_group.name}`) //DEBUG
         const condition: RegExp = new RegExp(approval_group.condition, "gm")
 
-        if (checkCondition(approval_group.check_type, condition, pr_diff_body, pr_files_list)) {
+        if (
+          checkCondition(
+            approval_group.check_type,
+            condition,
+            pr_diff_body,
+            pr_files_list,
+          )
+        ) {
           CUSTOM_REVIEW_REQUIRED = true
           // Combine users and team members in `approvers` list, excluding pr_owner
-          var approvers: string[] = []
-          await combineUsersTeams(octokit, context, organization, pr_owner, approval_group.users, approval_group.teams).then(value => {
-            approvers = value
-          })
-
+          var allApprovers: string[] = await combineUsersTeams(
+            octokit,
+            context,
+            organization,
+            pr_owner,
+            approval_group.users,
```

Comment:
```suggestion
            (approval_group.users) ? approval_group.users : [],
```
In case if approval group from the configuration file contains no users specified, `for` loop in function `combineUsersTeams` will fail as `approval_group.users` will pass `null` instead of an empty array. Previously this was checked with the `if (users) {}` condition inside the function. But this way seems to be more elegant.

Reply:
The program is type-checking fine currently, which means according to the type information it is incorrect to say `approval_group.users` can be null. That means the types should be fixed before applying the suggestion. I'll do that before getting to this.

Reply:
866181127d1969958569134f775a6f460928f962 adds validation and fixes the typing. I'll test this as soon as I'm free.

Reply:
Tested. Works as expected.